Executables/Game/AffectController.cs

```diff
@@ -0,0 +1,134 @@
+﻿using Dapper.Contrib.Extensions;
+using Microsoft.Extensions.Logging;
+using QuantumCore.API.Game.World;
+using Dapper;
+using System.Data;
+using Affect = QuantumCore.Database.Affect;
+using AffectAPI = QuantumCore.API.Core.Models.Affect;
+using QuantumCore.Game.Packets.Affects;
+
+namespace QuantumCore.Game
+{
+    public class AffectController : IAffectController
+    {
+        private readonly ILogger<IAffectController> _logger;
+        private readonly IDbConnection _db;
+
+        public AffectController(ILogger<IAffectController> logger, IDbConnection db) {
+            _logger = logger;
+            _db = db;
+        }
+        public void SendAffectAddPacket(IPlayerEntity playerEntity, Affect affect, int duration)
+        {
+            var affectAdd = new AffectAdd();
+            var affectAddPacket = new AffectAddPacket
+            {
+                Type = (uint) affect.Type,
+                ApplyOn = affect.ApplyOn,
+                ApplyValue = (uint) affect.ApplyValue,
+                Duration = (uint) duration,
+                Flag = (uint) affect.Flag,
+                SpCost = (uint) affect.SpCost
+            };
+            affectAdd.Elem[0] = affectAddPacket;
+            playerEntity.Connection.Send(affectAdd);
+        }
+
+        public async Task SendAffectRemovePacket(IPlayerEntity playerEntity, long type, byte applyOn)
+        {
+            await _db.QueryAsync("DELETE FROM affects WHERE PlayerId=@PlayerId and Type=@Type and ApplyOn=@ApplyOn", 
+                new { PlayerId = playerEntity.Player.Id, Type = type, ApplyOn = applyOn });
+            var affectRemovePacket = new AffectRemove
+            {
+                Type = (uint) type,
+                ApplyOn = applyOn,
+            };
+            await playerEntity.Connection.Send(affectRemovePacket);
+        }
+
+        public async Task AddAffect(IPlayerEntity playerEntity, int type, int applyOn, int applyValue, int flag, int duration, int spCost)
+        {
+            _logger.LogDebug("Add affect starting!");
+            _logger.LogDebug("::AddAffect Type:{Type}, ApplyOn:{ApplyOn}, ApplyValue:{ApplyValue}, Flag:{Flag}, Duration:{Duration}, SpCost:{SpCost}", type, applyOn, applyValue, flag, duration, spCost);  
```

Comment:
It might be easier to log via `_logger.LogDebug("Adding affect: {@Affect}", affect)`